src/views/List.jsx

```diff
@@ -15,6 +17,19 @@ export function List({ data, listToken }) {
 		}
 	}, [data, itemSearch]);
 
+	useEffect(() => {
+		async function sortShoppingList() {
+			const sortedData = await comparePurchaseUrgency(searchedData);
+			try {
+				setNewData(sortedData);
+			} catch (e) {
+				console.log('error sorting function: ', e);
```

Comment:
Let's put a message on the page rather than a console log, so we can keep the user in the loop when there are errors!

Reply:
To save you some time, here's one way you could handle putting an error message on the page:
In List.jsx:

`const [sortErrorMessage, setSortErrorMessage] = useState('');`

and 

```
useEffect(() => {
		...
			try {
				setNewData(sortedData);
			} catch (e) {
				setSortErrorMessage('There was an error sorting your list')
			}
    ...
```

and finally, somewhere inside of return()...

```
{sortErrorMessage && (
  <p> {sortErrorMessage} </p>
)}
```

Reply:
I agree it would make sense to display the message. Thank you for the suggestion and the example! 🤩